apps/wing/src/commands/compile.ts

```diff
@@ -67,39 +68,40 @@ function resolveSynthDir(outDir: string, entrypoint: string, target: Target) {
  * @returns the output directory
  */
 export async function compile(entrypoint: string, options: ICompileOptions): Promise<string> {
+  // create a unique temporary directory for the compilation
+  const tmpTargetdir = await mkdtemp(join(os.tmpdir(), "wing-"));
   const targetdir = join(dirname(entrypoint), "target");
   const wingFile = entrypoint;
   log("wing file: %s", wingFile);
   const wingDir = dirname(wingFile);
   log("wing dir: %s", wingDir);
+  const tmpSynthDir = resolveSynthDir(tmpTargetdir, wingFile, options.target);
```

Comment:
Intuitively I feel that perhaps we would want to create the temporary directory still under `target/xxx` (maybe `target/xxx.wsim.tmp`), so that npm dependencies will resolve relative to the entrypoint. @Chriscbr @ainvoner WDYT?

Reply:
Sounds like a good callout, definitely might prevent some nasty bugs in the future. 👍

Reply:
Looks like this will be fixed in #1929